File: 1385.两个数组间的距离值.py
#
# @lc app=leetcode.cn id=1385 lang=python3
#
# [1385] 两个数组间的距离值
#

# @lc code=start
class Solution:
    def findTheDistanceValue(self, arr1: List[int], arr2: List[int], d: int) -> int:
        def twoSideCheck(num):
            for i in range(0, d + 1):
                if num + i in num_dict or num - i in num_dict:
                    return False
            return True
        num_dict = {}
        for num in arr2:
            if num not in num_dict:
                num_dict[num] = 1
            else:
                num_dict[num] += 1
        res = 0
        for num in arr1:
            if twoSideCheck(num):
                res += 1
        return res
# @lc code=end

File: 1385.两个数组间的距离值.go
/*
 * @lc app=leetcode.cn id=1385 lang=golang
 *
 * [1385] 两个数组间的距离值
 */

// @lc code=start
func findTheDistanceValue(arr1 []int, arr2 []int, d int) (res int) {
    numMap := map[int]bool{}
    for _, num := range arr2 {
        numMap[num] = true
    }
    var twoSideCheck func(num int) bool
    twoSideCheck = func(num int) bool {
        for i := 0; i <= d; i++ {
            if numMap[num + i] || numMap[num - i] {
                return false
            }
        }
        return true
    }
    for _, num := range arr1 {
        if twoSideCheck(num) {
            res++
        }
    }
    return
}
// @lc code=end

